internal-api/src/main/java/datadog/trace/bootstrap/config/provider/stableconfig/StableConfig.java

```diff
@@ -18,9 +18,11 @@ public final class StableConfig {
   public StableConfig(Object yaml) {
     Map<Object, Object> map = (Map<Object, Object>) yaml;
     this.configId = map.get("config_id") == null ? null : String.valueOf(map.get("config_id"));
-    this.apmConfigurationDefault =
-        unmodifiableMap(
-            (Map<String, Object>) map.getOrDefault("apm_configuration_default", emptyMap()));
+    
+    // getOrDefault returns null if key exists with null value, so we need explicit null check
+    Map<String, Object> apmConfigDefault = (Map<String, Object>) map.getOrDefault("apm_configuration_default", emptyMap());
```

Comment:
Since we're now always checking for `null` we could simplify this line and just use `map.get`

```suggestion
    Map<String, Object> apmConfigDefault = (Map<String, Object>) map.get("apm_configuration_default");
```